tests/unit/output/test_load.py

```diff
@@ -0,0 +1,115 @@
+import pytest
+
+from dvc import output
+from dvc.output import LocalOutput, S3Output
+from dvc.stage import Stage
+
+
+@pytest.mark.parametrize(
+    "out_type,type_test_func",
+    [
+        ("outs", lambda o: not (o.metric or o.plot)),
+        ("metrics", lambda o: o.metric and not o.plot),
+        ("plots", lambda o: o.plot and not o.metric),
+    ],
+    ids=("outs", "metrics", "plots"),
+)
+def test_load_from_pipeline(dvc, out_type, type_test_func):
+    outs = output.load_from_pipeline(
+        Stage(dvc),
+        [
+            "file1",
+            "file2",
+            {"file3": {"cache": True}},
+            {},
+            {"file4": {"cache": False}},
+            {"file5": {"persist": False}},
+            {"file6": {"persist": True, "cache": False}},
+        ],
+        out_type,
+    )
+    cached_outs = {"file1", "file2", "file3", "file5"}
+    persisted_outs = {"file6"}
+    assert len(outs) == 6
+
+    for i, out in enumerate(outs, start=1):
+        assert isinstance(out, LocalOutput)
+        assert out.def_path == f"file{i}"
+        assert out.use_cache == (out.def_path in cached_outs)
+        assert out.persist == (out.def_path in persisted_outs)
+        assert out.info == {}
+        assert type_test_func(out)
+
+
+def test_load_from_pipeline_accumulates_flag(dvc):
+    outs = output.load_from_pipeline(
+        Stage(dvc),
+        [
+            "file1",
+            {"file2": {"cache": False}},
+            {"file1": {"persist": False}},
+            {"file2": {"persist": False}},
```

Comment:
`persist` = `False` is default, right? Maybe for the accumulation test, it would be better to make it `True` and assert that `file2` is not cached and persisted? That way we are sure we are not using defaults.

Reply:
That's a very good point. Fixed.